adders/prettier/config/adder.ts

```diff
@@ -0,0 +1,128 @@
+            content: ({ data, dependencies }) => {
+                data.scripts ??= {};
+                const scripts: Record<string, string> = data.scripts;
+                const CHECK_CMD = "prettier --check .";
+                scripts["format"] ??= "prettier --write .";
+                scripts["format:check"] ??= CHECK_CMD;
+
+                if (hasEslint(dependencies)) {
+                    scripts["lint"] ??= `${CHECK_CMD} && eslint .`;
+                    if (!scripts["lint"].includes(CHECK_CMD)) scripts["lint"] += ` && ${CHECK_CMD}`;
+                }
+            },
+        },
+        {
+            name: () => "eslint.config.js",
+            contentType: "script",
+            condition: ({ dependencies }) => hasEslint(dependencies),
+            content: ({ ast, imports, exports, common }) => {
+                // TODO: maybe this could be more intelligent and we can detect the name of the default import?
+                imports.addDefault(ast, "eslint-plugin-svelte", "svelte");
+                imports.addDefault(ast, "eslint-config-prettier", "prettier");
+
+                const fallbackConfig = common.expressionFromString("[]");
+                const defaultExport = exports.defaultExport(ast, fallbackConfig);
+                const array = defaultExport.value;
+                if (array.type !== "ArrayExpression") return;
+
+                const prettier = common.expressionFromString("prettier");
+                const sveltePrettierConfig = common.expressionFromString("svelte.configs['flat/prettier']");
+                const configSpread = common.createSpreadElement(sveltePrettierConfig);
+
+                const nodesToInsert = [];
+                if (!common.hasNode(array, prettier)) nodesToInsert.push(prettier);
+                if (!common.hasNode(array, configSpread)) nodesToInsert.push(configSpread);
+
+                // finds index of `...svelte.configs["..."]`
+                const idx = array.elements.findIndex(
+                    (el) =>
+                        el?.type === "SpreadElement" &&
+                        el.argument.type === "MemberExpression" &&
+                        el.argument.object.type === "MemberExpression" &&
+                        el.argument.object.property.type === "Identifier" &&
+                        el.argument.object.property.name === "configs" &&
+                        el.argument.object.object.type === "Identifier" &&
+                        el.argument.object.object.name === "svelte",
+                );
+
+                if (idx !== -1) {
+                    array.elements.splice(idx + 1, 0, ...nodesToInsert);
+                } else {
+                    // append to the end as a fallback
+                    array.elements.push(...nodesToInsert);
+                }
+            },
+        },
+    ],
+});
+
+function hasEslint(deps: Record<string, string>): boolean {
+    return !!deps["eslint"] && deps["eslint"].startsWith("9");
```

Comment:
What would we do if a user has v8 installed?
Should we maybe make this a pre-requisite and hint him to upgrade his eslint first?

Reply:
> What would we do if a user has v8 installed?

it just won't modify the config if it's v8. we should only be targeting v9 since the flat config is the new default for it. having to account for both config formats is something we definitely don't want to support 😄

> Should we maybe make this a pre-requisite and hint him to upgrade his eslint first?

I don't think we need a pre-req for this. although, a post-execution log could just say that "eslint v9 or higher wasn't 
 detected and X step was skipped" or something along those lines.

Reply:
That sounds good to me. Though "eslint v9 or higher wasn't detected and X step was skipped" might not be quite the right wording as it would work just fine when there's no eslint installed. Instead I think you'd have to say "An older major version of eslint was detected and step X was skipped"

Reply:
How about this for the warning?:
![img](https://i.imgur.com/Dg8RxjA.png)

Leaving out the eslint version does feel a bit off, how would the user know which one to upgrade to?

Reply:
that looks great. I think the message is fine as you have it because I would assume that means you should use the latest version. But if you wanted to include the versions you could say something like "eslint 8 was detected, but eslint 9+ is required."